module/VuFind/src/VuFind/ILS/Driver/PAIA.php

```diff
@@ -726,20 +726,24 @@ public function getMyHolds($patron)
      */
     public function getMyProfile($patron)
     {
-        //todo: read VCard if avaiable in patron info
-        //todo: make fields more configurable
         if (is_array($patron)) {
+            $type = null;
+            if (is_array($patron['type'])) {
+                $type = implode(', ', $patron['type']);
+            } elseif (isset($patron['type'])) {
+                $type = $patron['type'];
+            }
             return [
-                'firstname'      => $patron['firstname'],
-                'lastname'       => $patron['lastname'],
-                'address1'       => null,
-                'address2'       => null,
-                'city'           => null,
-                'country'        => null,
-                'zip'            => null,
-                'phone'          => null,
-                'mobile_phone'   => null,
-                'group'          => null,
+                'firstname'  => $patron['firstname'],
+                'lastname'   => $patron['lastname'],
+                'address1'   => $patron['address'],
+                'address2'   => null,
+                'city'       => null,
+                'country'    => null,
+                'zip'        => null,
+                'phone'      => null,
+                'mobile_phone' => null,
+                'group'      => $this->getReadableGroupType($type),
```

Comment:
Rather than calling this down here, where it may receive a comma-delimited list from above, what if you instead simplified the block that defines $type to something like:

<pre>
$type = isset($patron['type'])
    ? implode(',', array_map([$this, 'getReadableGroupType'], (array)$patron['type']))
    : null;
</pre>

I think that covers all the same cases in fewer lines of code and also means that the mapping routine only needs to deal with one type value at a time.